database/models.py

```diff
@@ -71,3 +72,22 @@ def to_json(self, *args, **kwargs):
         obj["_id"] = str(obj["_id"])
         obj["author"] = {"name": self.author.name, "id": str(self.author.id)}
         return obj
+
+
+class TaskStatus(Enum):
+    """Possible Task Statuses"""
+
+    NOT_STARTED = {"quotes": "NOT_STARTED", "author": "NOT_STARTED"}
+    QUOTE_IN_PROGRESS = {"quotes": "IN_PROGRESS", "author": "NOT_STARTED"}
+    AUTHOR_IN_PROGRSS = {"quotes": "SUCCESS", "author": "IN_PROGRESS"}
+    SUCCESS = {"quotes": "SUCCESS", "author": "SUCCESS"}
```

Comment:
Enums should contain constants such as int, str and not the dictionary. You can use them like - 
```suggestion
class TaskStatus(Enum):
    """Task Statuses"""

    NOT_STARTED: str = "NOT_STARTED"
    IN_PROGRESS: str = "IN_PROGRESS"
    FAILED: str = "FAILED"
    SUCCESS: str = "SUCCESS"
```

Reply:
But then I can't store result as dict type.

Reply:
I will need to use StringField

Reply:
can't store like you said '{quotes: SUCCESS, author: IN_PROGRESS}'